### ISSUE TITLE
Error: Cannot find module @babel

### ISSUE THREAD

**Issue by rejas** [eladnava/tldrlegal#9]
Got this error when running against my create-react-app based project, any idea why it crashes?

```
C:\Projekte\mine>tldrlegal
module.js:550
    throw err;
    ^

Error: Cannot find module 'C:\Projekte\mine/node_modules/@babel/plugin-proposal-class-properties/node_modules/@babel/package.json'
    at Function.Module._resolveFilename (module.js:548:15)
    at Function.Module._load (module.js:475:25)
    at Module.require (module.js:597:17)
    at require (internal/module.js:11:18)
    at getPackage (C:\Users\rejas\AppData\Roaming\npm\node_modules\tldrlegal\node_modules\legally\lib\legally.js:17:13)
    at packages.reduce (C:\Users\rejas\AppData\Roaming\npm\node_modules\tldrlegal\node_modules\legally\lib\legally.js:36:8)
    at Array.reduce (<anonymous>)
    at module.exports.root (C:\Users\rejas\AppData\Roaming\npm\node_modules\tldrlegal\node_modules\legally\lib\legally.js:34:19)
    at Object.<anonymous> (C:\Users\rejas\AppData\Roaming\npm\node_modules\tldrlegal\index.js:28:16)
    at Module._compile (module.js:653:30)

```

**Comment by eladnava** [comment]
Hi @rejas,
Which version of Node are you running?

Check with `node -v`.

**Comment by rejas** [comment]
v8.15.1

**Comment by eladnava** [comment]
Seems like possibly a module missing which is defined in your `package.json` but not installed in `node_modules`? 

Try running `npm install` in your project root and then try `tldrlegal` again.

**Comment by rejas** [comment]
I did npm install multiple times without getting a better result :-(

**Comment by eladnava** [comment]
Hi @rejas,
I cannot reproduce, I just ran
```
create-react-app my-app
cd my-app
tldrlegal
```

And the result:

![Screen Shot 2019-05-21 at 7 44 17 PM](https://user-images.githubusercontent.com/4394189/58077539-db252680-7c00-11e9-8fc3-9429a5129463.png)

Running the latest `create-react-app` (3.0.1) and `tldrlegal` (1.0.6) using Node v8.9.0. Can you make sure your versions of `tldrlegal` and `create-react-app` are the latest?

**Comment by rejas** [comment]
Yes, I did use that latest versions. But running even the basic example you jut gave also triggers an error on my side. The only differences I see right now are node version (8.9.0 vs 8.15.1) and os (macos vs win10) ...

**Comment by eladnava** [comment]
@rejas Just tested with Node `v8.15.1` and it worked as expected, so most likely the issue is related to Windows file path differences?

It does appear to be an upstream issue with [`legally`](https://github.com/franciscop/legally) though as per the stacktrace:
```
Error: Cannot find module 'C:\Projekte\mine/node_modules/@babel/plugin-proposal-class-properties/node_modules/@babel/package.json'
    at Function.Module._resolveFilename (module.js:548:15)
    at Function.Module._load (module.js:475:25)
    at Module.require (module.js:597:17)
    at require (internal/module.js:11:18)
    at getPackage (C:\Users\rejas\AppData\Roaming\npm\node_modules\tldrlegal\node_modules\legally\lib\legally.js:17:13)
    at packages.reduce (C:\Users\rejas\AppData\Roaming\npm\node_modules\tldrlegal\node_modules\legally\lib\legally.js:36:8)
    at Array.reduce (<anonymous>)
```

Can you simply try installing `legally` and see if it runs on your project?

```
npm install legally -g 
legally
```

**Comment by rejas** [comment]
Nope, legally runs through fine:

```Working on ".". It might take a while...


┌──────────────────────────────────────────────────────────────────────────────┐
│                                                                              │
│                               Packages (1766)                                │
│                                                                              │
├───────────────────────────┬────────────────┬────────────────┬────────────────┤
│ Module name               │ package        │ License        │ README         │
├───────────────────────────┼────────────────┼────────────────┼────────────────┤
│ @babel/code-frame@7.0.0   │ MIT            │ MIT            │ -              │
....
```

**Comment by lamuertepeluda** [comment]
Same issue here, I guess (macOS Mojave, latest)

```bash
tldrlegal
internal/modules/cjs/loader.js:583
    throw err;
    ^

Error: Cannot find module '/Users/vito/Development/JavaScript/IOC/mobile-app/node_modules/@turf/transform-scale/node_modules/@turf/package.json'
    at Function.Module._resolveFilename (internal/modules/cjs/loader.js:581:15)
    at Function.Module._load (internal/modules/cjs/loader.js:507:25)
    at Module.require (internal/modules/cjs/loader.js:637:17)
    at require (internal/modules/cjs/helpers.js:22:18)
    at getPackage (/Users/vito/.nvm/versions/node/v10.15.1/lib/node_modules/tldrlegal/node_modules/legally/lib/legally.js:17:13)
    at packages.reduce (/Users/vito/.nvm/versions/node/v10.15.1/lib/node_modules/tldrlegal/node_modules/legally/lib/legally.js:36:8)
    at Array.reduce (<anonymous>)
    at module.exports.root (/Users/vito/.nvm/versions/node/v10.15.1/lib/node_modules/tldrlegal/node_modules/legally/lib/legally.js:34:19)
    at Object.<anonymous> (/Users/vito/.nvm/versions/node/v10.15.1/lib/node_modules/tldrlegal/index.js:28:16)
    at Module._compile (internal/modules/cjs/loader.js:689:30)
```

I think this is an issue with `@` repositories such as `@babel` or `@turf` which contain subfolders...

I can confirm this does not happen with `legally`

**Comment by eladnava** [comment]
Not entirely sure how I can reproduce this, I just ran the following:

```
nvm use v8.15.1
npm install @turf/transform-scale --save
tldrlegal
```

It ran successfully and correctly inspected `transform-scale` and all of its dependencies as well. I'm running macOS High Sierra 10.13.6. Do you think that has to do with it?

**Comment by lamuertepeluda** [comment]
@eladnava sorry I omitted that I was using node v10.15.1, macOS Mojave 10.14.5
However after reading your message I tried with v8.15.1 (and 8.12.0 and 9.11.1) and I get the same error...

The fact is that the error message says it cannot find module `<SRC>/node_modules/@turf/projection/node_modules/@turf/package.json`

or for some other repositories

`<SRC>/node_modules/@babel/helper-replace-supers/node_modules/@babel/package.json`

where <SRC> is the folder where I launched the command. Looks like something to do with the way this utility performs directory walking, I guess?

**Comment by eladnava** [comment]
@lamuertepeluda From the working directory in which you ran `tldrlegal` from, can you please check if the file is reachable via running this command?

```
cat ./node_modules/@turf/projection/node_modules/@turf/package.json
```

**Comment by lamuertepeluda** [comment]
@eladnava no, it isn't, since `@turf` is only a directory with all `@turf/XXX` modules.

```bash
$cat ./node_modules/@turf/projection/node_modules/@turf/package.json
cat: ./node_modules/@turf/projection/node_modules/@turf/package.json: No such file or directory

$ls ./node_modules/@turf
area                     boolean-within           destination              projection
bbox                     buffer                   distance                 rhumb-bearing
bbox-polygon             center                   envelope                 rhumb-destination
boolean-contains         centroid                 helpers                  rhumb-distance
boolean-equal            circle                   intersect                transform-scale
boolean-point-in-polygon clean-coords             invariant
boolean-point-on-line    clone                    meta
```

**Comment by rejas** [comment]
@eladnava same here in my project. the file doesnt exist since @babel is just the directory with all packages from the babel organization.

**Comment by rejas** [comment]
@eladnava any ideas what might cause the issue? maybe npm version? mine is 6.4.1 for whatsever its worth...

**Comment by eladnava** [comment]
@rejas Unfortunately I just tried with `npm` version `6.4.1` and it still works on my side.

```
$ node -v
v10.1.0
$ npm -v
6.4.1
```

Ran the following commands:
```
npm install -g create-react-app tldrlegal
create-react-app my-app
cd my-app
tldrlegal
```

Also tried with a fresh project with just a scoped package:
```
npm install @turf/transform-scale --save
tldrlegal
```

All invocations worked as expected. Any chance one of you experiencing this issue can try debugging this on your own machine to see why this error is cropping up?

**Comment by rejas** [comment]
My guts say it has something to do with how npm installs packages. In the screenshot attached on the left you can see a snippet of the node_modules directory fresh my-app (like you created in the post before). On the right you see the same snippet with my app that has problems running the legal plugin.
![legal](https://user-images.githubusercontent.com/1313547/59017072-fa4bc700-8842-11e9-90cb-c30221dc31a8.png)

Now, when I first created the my-app example, tldrlegal gave the same error as always. Then I deleted the whole node_modules directory and ran `npm install`  again. Then it worked magically. Thats why I think it has something to do with npm and how it installs packages and maybe leaves behind some bogus node_modules directory. 


Anyway, it would be better though if legal would handle those occurences where a bogus package-json cannot be found. Then again, legally is already a few version ahead, do you have plans on upgrading your library to use the latest legally @eladnava  ?

**Comment by eladnava** [comment]
@rejas Interesting findings. So we could deduce that `create-react-app` is creating malformed `node_modules` or a format which `tldrlegal` does not cope with currently. 

Can you please confirm @rejas if running `legally` on the fresh-created `create-react-app` project runs even though `tldrlegal` errors?

```
create-react-app my-app
cd my-app
tldrlegal # error

npm install legally -g 
legally # no error?
```

**Comment by rejas** [comment]
Yes, I can confirm that @eladnava

**Comment by rejas** [comment]
But it could also be a problem with `yarn`since create-react-app uses that for its initital setup. Just tested it by 
deleting node_modules, running `npm install` -> tldrlegal runs fine
deleting node_modules (and lock file), running `yarn install`-> tldrlegal gives error

**Comment by eladnava** [comment]
@rejas Yes you got it!

I didn't have `yarn` installed so that explains why my `create-react-app` invocation wasn't creating a broken `node_modules/` directory structure that `tldrlegal` wasn't liking!

All along that was the difference causing this strange behavior.

Updating `tldrlegal` to use the latest version of `legally` in https://github.com/eladnava/tldrlegal/commit/1d3ebed55725b0b6e87de3fd0a3f3b16c6248999 has added support for Yarn's special `node_modules/` format.

Published `tldrlegal` version `1.0.8` which resolves this. 

Thanks all for your help! 🎉

**Comment by rejas** [comment]
Nice. Works indeed! Thanks for your help too!
